quizbizz/src/pages/authentication/login/Loginform.jsx

```diff
@@ -30,59 +29,55 @@ const Loginform = () => {
   };
   const loginUser = async () => {
     if (formDetails.email === "" || formDetails.password === "") {
-      // showToast("Please Enter the details first", "ERROR");
+      showToast("Please Enter the details first", "ERROR");
       return;
     }
-    try {
-      const response = await Axios.post("/api/auth/login", {
+
+    const { data, success, message, statusCode } = await postApi(
+      "/api/auth/login",
+      {
         email: formDetails.email,
         password: formDetails.password,
-      });
+      },
+      false
+    );
 
-      // console.log(response);
-      const token = response.data.encodedToken;
+    if (success && statusCode === 200) {
+      const token = data.encodedToken;
       localStorage.setItem("token", token);
-
-      // dispatchUser({
-      //   type: "LOGIN",
-      //   payload: { value: response.data.foundUser },
-      // });
-
+      dispatchUser({
+        type: "LOGIN",
+        payload: { value: data.foundUser },
+      });
+      showToast("Login successfull", "SUCCESS");
       navigate("/");
-    } catch (error) {
-      console.error(error);
+    } else {
+      showToast("Something Went wrong", "ERROR");
     }
   };
 
   const loginAsGuest = async (e) => {
     e.preventDefault();
-    try {
-      const response = await Axios.post("/api/auth/login", {
+    const { data, success, message, statusCode } = await postApi(
+      "/api/auth/login",
+      {
         email: "adarshbalika@gmail.com",
         password: "adarshBalika123",
-      });
+      },
+      false
+    );
 
-      console.log(response);
-      const token = response.data.encodedToken;
+    if (success && statusCode === 200) {
+      const token = data.encodedToken;
       localStorage.setItem("token", token);
-
-      // dispatchUser({
-      //   type: "LOGIN",
-      //   payload: { value: response.data.foundUser },
-      // });
-
-      // dispatchCart({
-      //   type: "SET_CART",
-      //   payload: {
-      //     value: response.data.foundUser.cart,
-      //   },
-      // });
-
-      //TODO set wishlist here
-
+      dispatchUser({
+        type: "LOGIN",
+        payload: { value: data.foundUser },
+      });
+      showToast("Login successfull", "SUCCESS");
       navigate("/");
-    } catch (error) {
-      console.error(error);
+    } else {
```

Comment:
suggestion: on login as guest just fill the form with demo credentials. so that you do not have to create separate function for loginas guest

Reply:
I have kept it like this because in this user can login at a click, but the approach you are talking about the user has to click the button twice one for filling the form and another time for login, that is why i have kept it like this